src/CurrencyInput/index.jsx

```diff
@@ -0,0 +1,231 @@
+         */
+        spellCheck   : PropTypes.bool,
+        /**
+         *  Input text alignment
+         */
+        textAlign    : PropTypes.oneOf( [ 'left', 'right' ] ),
+        /**
+         *  Input string value
+         */
+        value        : PropTypes.string,
+    };
+
+    static defaultProps =
+    {
+
+        autoCapitalize : undefined,
+        autoComplete   : undefined,
+        autoCorrect    : undefined,
+        className      : undefined,
+        cssMap         : undefined,
+        currency       : undefined,
+        hasError       : false,
+        id             : undefined,
+        isDisabled     : false,
+        isReadOnly     : false,
+        onBlur         : undefined,
+        onChange       : undefined,
+        onClick        : undefined,
+        onClickIcon    : undefined,
+        onFocus        : undefined,
+        onKeyDown      : undefined,
+        onKeyPress     : undefined,
+        onKeyUp        : undefined,
+        onMouseOut     : undefined,
+        onMouseOver    : undefined,
+        placeholder    : undefined,
+        spellCheck     : undefined,
+        textAlign      : 'left',
+        value          : '',
+    };
+
+
+    static displayName = 'CurrencyInput';
+
+    constructor( props )
+    {
+        super( props );
+        this.state = {
+            valueFormatted : ''
+        };
+        this.handleBlur   = this.handleBlur.bind( this );
+        this.handleChange = this.handleChange.bind( this );
+    }
+
+    handleBlur()
+    {
+        const newVal = Number( this.state.valueFormatted.replace( /[^0-9\.-]/g, '' ) );
+        this.setState( {
+            valueFormatted: newVal
+        } );
+        if ( typeof this.props.onBlur === 'function' )
+        {
+            this.props.onBlur();
+        }
+    }
+
+    handleChange( { value } )
+    {
+        this.setState( { valueFormatted: value } );
+        if ( typeof this.props.onChange === 'function' )
+        {
+            this.props.onChange( { value } );
+        }
+    }
+
+    render()
+    {
+
+        const {
+            autoCapitalize,
+            autoComplete,
+            autoCorrect,
+            cssMap = createCssMap( this.context.TextInput, this.props ),
+            currency,
+            id = generateId( 'CurrencyInput' ),
+            isDisabled,
+            isReadOnly,
+            placeholder,
+            spellCheck,
+        } = this.props;
+
+        return (
+            <TextInput
+                { ...this.props }
+                autoCapitalize = { autoCapitalize }
+                autoComplete   = { autoComplete }
+                autoCorrect    = { autoCorrect }
+                className      = { cssMap.main }
+                disabled       = { isDisabled }
+                id             = { id }
+                placeholder    = { placeholder }
+                readOnly       = { isReadOnly }
+                ref            = { this.inputRef }
```

Comment:
`this.inputRef` does not exist. let’s just remove this for now and use `forwardRef` when we convert this component to hooks